core/common/src/Clock.kt

```diff
@@ -53,5 +53,19 @@ private class InstantTimeMark(private val instant: Instant, private val clock: C
     override fun minus(duration: Duration): TimeMark = InstantTimeMark(instant - duration, clock)
 }
 
+/**
+ * Returns the [Clock] by storing an initial [TimeMark] using [TimeSource.markNow] and [returns][Clock.now] the elapsed
+ * time using [TimeMark.elapsedNow] plus the provided [offset].
+ *
+ * This clock stores the initial [TimeMark], so repeatedly creating [Clock]s from the same [TimeSource] results
+ * into different [Instant]s iff the time of the [TimeSource] was increased. To sync different [Clock]s, use the [offset]
+ * parameter.
+ */
```

Comment:
Could you also add a `@sample`? You can take almost any other function as an example.